app.py

```diff
@@ -209,7 +209,34 @@ def _get_app_path(self, version=None):
 
     def _launch_app(self, context, file_to_open=None, version=None):
         """
-        Launches an app
+        Launches an application. No environment variable change is leaked to the outside world.
+        :param context: Tank context we will opening the app in.
+        :param file_to_open: Optional file to open when the app launches. Can be None.
+        :param version: Version of the app to launch. Specifying None means the default version will
+                        be picked.
+        """
+        try:
+            # Clone the environment variables
+            environ_clone = os.environ.copy()
```

Comment:
It might be nice to move this piece of code to a decorator in core. I have an inkling it might be reused elsewhere.